app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,200 @@
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("createTask 메소드는")
+    class Describe_createTask {
+        @Nested
+        @DisplayName("만약 등록할 Task가 주어진다면")
+        class Context_with_task {
+            @Test
+            @DisplayName("Task를 생성하고, 리턴한다.")
+            void it_create_task_return_task() {
+                int sizeBeforeCreation = getTasksSize();
+                Task task = getTask();
+
+                Task createdTask = taskService.createTask(task);
+                int sizeAfterCreation = getTasksSize();
+
+                assertThat(createdTask.getTitle()).isEqualTo(task.getTitle());
+                assertThat(sizeAfterCreation - sizeBeforeCreation).isEqualTo(1);
+            }
+        }
+
+        @Nested
+        @DisplayName("만약 등록할 Task가 주어진다면")
+        class Context_without_task {
+            @Test
+            @DisplayName("NullPointerException 을 리턴한다.")
+            void it_return_exception() {
+                assertThatThrownBy(() -> taskService.createTask(null)).isInstanceOf(NullPointerException.class);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("updateTask 메소드는")
+    class Describe_updateTask {
+        @Nested
+        @DisplayName("만약 등록된 Task의 id 와 수정할 Task 가 주어진다면")
+        class Context_with_id_and_task {
+            @Test
+            @DisplayName("해당 id의 Task를 수정하고, 리턴한다.")
+            void it_update_task_return_task() {
+                Task source = getTaskWithPostfix();
+
+                Task updatedTask = taskService.updateTask(VALID_ID, source);
+
+                assertThat(updatedTask.getTitle()).isEqualTo(source.getTitle());
+            }
+        }
+
+        @Nested
+        @DisplayName("만약 등록된 Task의 id 만 주어진다면")
+        class Context_with_id {
+            @Test
+            @DisplayName("NullPointerException 이 리턴한다.")
+            void it_update_task_return_task() {
+                assertThatThrownBy(() -> taskService.updateTask(VALID_ID, null)).isInstanceOf(NullPointerException.class);
+            }
+        }
+
+        @Nested
+        @DisplayName("만약 등록된 Task 만 주어진다면")
+        class Context_with_task {
+            @Test
+            @DisplayName("TaskNotFoundException 가 리턴한다.")
+            void it_return_taskNotFoundException() {
+                Task source = getTaskWithPostfix();
+
+                assertThatThrownBy(() -> taskService.updateTask(null, source)).isInstanceOf(TaskNotFoundException.class);
+            }
+        }
+
+        @Nested
+        @DisplayName("만약 등록되지 않은 Task의 id 와 Task가 있다면 ")
+        class Context_with_invalid_id_and_task {
+            @Test
+            @DisplayName("TaskNotFoundException 가 리턴한다.")
+            void it_return_taskNotFoundException() {
+                Task source = getTask();
+
+                assertThatThrownBy(() -> taskService.updateTask(INVALID_ID, source)).isInstanceOf(TaskNotFoundException.class);
```

Comment:
고민되는 부분이 하나 있는데요. 
```suggestion
                assertThatThrownBy(() -> taskService.updateTask(INVALID_ID, getTask())).isInstanceOf(TaskNotFoundException.class);
```
윗 부분을 이런 식으로 표현하는게 좋을까요?
메소드를 삽입하면 햇갈릴거 같기도 하고, 
```
 Task source = getTask();
```
이것만 달랑 있자하니 뭔가 애매하네요 ㅜ

Reply:
이 둘은 똑같습니다. 어느쪽이 더 이해하기 쉬운지를 보고 선택하셔야 해요.
이해하기 쉬운 것이 어떤 것인지 잘 모르겠다면 "3초 안에 이해할 수 있다면" 이해하기 쉬운 것이라고 생각해 보세요.
도움이 될 거에요.

Reply:
보다보니 후자가 더 괜찮을 것 같습니다. source에 Task 인스턴스를 넣고, 따라가는 것보다 한줄로 읽는게 간단하다고 느껴지네요.